jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/HibernateSearchProcessor.java

```diff
@@ -93,18 +81,20 @@ else if (!providerModule.equals(searchMapperModuleName)) {
 
         // use Search module name specified in persistence unit definition
         if (searchMapperModuleName != null && !IGNORE.equals(searchMapperModuleName)) {
-            ModuleIdentifier moduleIdentifier = ModuleIdentifier.fromString(searchMapperModuleName);
-            moduleSpecification.addSystemDependency(new ModuleDependency(moduleLoader, moduleIdentifier, false, true, true, false));
-            ROOT_LOGGER.debugf("added %s dependency to %s", moduleIdentifier, deploymentUnit.getName());
+            moduleSpecification.addSystemDependency(ModuleDependency.Builder.of(moduleLoader, searchMapperModuleName).
```

Comment:
I believe the searchMapperModuleName var has not been canonicalized. It holds a value that was read from PersistenceUnitMetadata.getProperties().getProperty(Configuration.HIBERNATE_SEARCH_MODULE) and I don't expect that value to have been canonicalized.

I was going to suggest a fix, but the only use of this is to pass it into ModuleDependency.Builder, and that handles the canonicalization.